high_card.rb

```diff
@@ -1,5 +1,35 @@
 # Use these two arrays to generate a deck of cards.
-ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
-suits = [ "hearts", "spades", "clubs", "diamonds" ]
+ranks = [2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K", "A"]
+suits = [ "clubs", "diamonds", "hearts", "spades" ]
 
+deck = []
+player = ""
 players = []
+#add a weight variable so that there are no ties. suits will have a weight.
+# clubs < diamonds < hearts < spades
+weight = 0
+ranks.each_with_index do |rank, i|
+  suits.each do |suit|
+    deck.push({value: rank, suit: suit, weight: weight})
+    weight += 1
+  end
+end
+
+deck.shuffle!
```

Comment:
Try to avoid modifying data structures as much as possible! When in doubt, reassign.